core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -197,11 +226,114 @@ contract PricelessPositionManager is Testable {
         tokenCurrency.burn(numTokens.rawValue);
     }
 
+    /**
+    * @notice After expiration of the contract the DVM is asked what for the prevailing price at the time of
+    * expiration.
+    */
+    function expire() public onlyPostExpiration() {
+        _requestOraclePrice(expirationTimestamp);
+    }
+
+    /**
+     * @notice After a contract has passed maturity all token holders can redeem their tokens for underlying at
+     * the prevailing price defined by the DVM from the `expire` function. This Burns all tokens from the caller
+     * of `tokenCurrency` and sends back the proportional amount of `collateralCurrency`.
+     */
+    function settleExpired() public onlyPostExpiration() {
+        // Get the current settlement price. If it is not resolved will revert.
+        FixedPoint.Unsigned memory settlementPrice = _getOraclePrice(expirationTimestamp);
+
+        // Get caller's tokens balance and calculate amount of underlying entitled to them.
+        FixedPoint.Unsigned memory tokensToRedeem = FixedPoint.Unsigned(tokenCurrency.balanceOf(msg.sender));
```

Comment:
So token holders must redeem all at once? Any tradeoffs to not allowing partial redemptions? FWIW, I like the simplicity

Reply:
ye, all or nothing. I thought about this one but didnt see a situation in which a token holder would want to redeem anything less than their current full balance. Open to changing it but this is defos easier